backend/app/core/auth.py

```
@@ -0,0 +1,135 @@
+    """
+    credentials_exception = HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="Could not validate credentials",
+        headers={"WWW-Authenticate": "Bearer"},
+    )
+    
+    try:
+        token = credentials.credentials
+        user_id = verify_token(token)
+        if user_id is None:
+            raise credentials_exception
```

Ruff (0.14.0)

Abstract `raise` to an inner function

(TRY301)